samcli/local/lambdafn/zip.py

```diff
@@ -19,6 +19,70 @@
 
 LOG = logging.getLogger(__name__)
 
+S_IFLNK = 0xA
+
+
+def _is_symlink(file_info):
+    """
+    Check the upper 4 bits of the external attribute for a symlink.
+    See: https://unix.stackexchange.com/questions/14705/the-zip-formats-external-file-attribute
+
+    Parameters
+    ----------
+    file_info : zipfile.ZipInfo
+        The ZipInfo for a ZipFile
+
+    Returns
+    -------
+    bool
+        A response regarding whether the ZipInfo defines a symlink or not.
+    """
+
+    return (file_info.external_attr >> 28) == 0xA
```

Comment:
Should this be using the `S_IFLNK` constant from above?

Why not use `stat. S_IFLNK` here instead?

Reply:
This is not looking at a file on the filesystem. It is looking at a number of bytes in the `file_info` entry, the External Attributes for the file. See [The zip format's external file attribute](https://unix.stackexchange.com/questions/14705/%20the-zip-formats-external-file-attribute) If you want to dig into it. But the short answer is because it isn't a file.

Reply:
I see. You made a constant `S_IFLNK = 0xA` above. Can you either remove it the constant or replace the `0xA` here with `S_IFLNK`. Otherwise, it looks like an unused constant.

Reply:
^ that was with regard to pythons `stat.S_ISLNK()` ... I have no good excuse for not comparing to the *S_IFLNK = 0xA* constant defined directly above the function. :-)